From c935fb42f0edeb5aca0f65a3af476edf5aba7a34 Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Thu, 23 Sep 2021 12:57:52 -0700
Subject: [PATCH] cryptodev-linux: update to 1.12

Remove upstream backport.

Use AUTORELEASE for simplicity.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
(cherry picked from commit 585cef5f1a9c1c3aecd7d231364618e96d03ab65)
---
 package/kernel/cryptodev-linux/Makefile       |  6 ++--
 .../010-fix-build-for-kernel-v5.9-rc1.patch   | 32 -------------------
 2 files changed, 3 insertions(+), 35 deletions(-)
 delete mode 100644 package/kernel/cryptodev-linux/patches/010-fix-build-for-kernel-v5.9-rc1.patch

diff --git a/package/kernel/cryptodev-linux/Makefile b/package/kernel/cryptodev-linux/Makefile
index 85065e4eeb6c..345036b59d0c 100644
--- a/package/kernel/cryptodev-linux/Makefile
+++ b/package/kernel/cryptodev-linux/Makefile
@@ -10,12 +10,12 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=cryptodev-linux
-PKG_VERSION:=1.11
-PKG_RELEASE:=1
+PKG_VERSION:=1.12
+PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE_URL:=https://codeload.github.com/$(PKG_NAME)/$(PKG_NAME)/tar.gz/$(PKG_NAME)-$(PKG_VERSION)?
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_HASH:=d71fd8dafc40147586f5bc6acca8fce5088d9c576d1142fe5aeb7b0813186a11
+PKG_HASH:=f51c2254749233b1b1d7ec9445158bd709f124f88e1c650fe2faac83c3a81938
 PKG_LICENSE:=GPL-2.0
 PKG_LICENSE_FILES:=COPYING
 
diff --git a/package/kernel/cryptodev-linux/patches/010-fix-build-for-kernel-v5.9-rc1.patch b/package/kernel/cryptodev-linux/patches/010-fix-build-for-kernel-v5.9-rc1.patch
deleted file mode 100644
index 09768a1f3aaf..000000000000
--- a/package/kernel/cryptodev-linux/patches/010-fix-build-for-kernel-v5.9-rc1.patch
+++ /dev/null
@@ -1,32 +0,0 @@
-From 2f5e08aebf9229599aae7f25db752f74221cd71d Mon Sep 17 00:00:00 2001
-From: Joan Bruguera <joanbrugueram@gmail.com>
-Date: Fri, 14 Aug 2020 00:13:38 +0200
-Subject: [PATCH] Fix build for Linux 5.9-rc1
-
-See also: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=64019a2e467a288a16b65ab55ddcbf58c1b00187
-          https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=bce617edecada007aee8610fbe2c14d10b8de2f6
-          https://lore.kernel.org/lkml/CAHk-=wj_V2Tps2QrMn20_W0OJF9xqNh52XSGA42s-ZJ8Y+GyKw@mail.gmail.com/
-
-Signed-off-by: Joan Bruguera <joanbrugueram@gmail.com>
----
- zc.c | 6 +++++-
- 1 file changed, 5 insertions(+), 1 deletion(-)
-
---- a/zc.c
-+++ b/zc.c
-@@ -76,10 +76,14 @@ int __get_userbuf(uint8_t __user *addr,
- 	ret = get_user_pages_remote(task, mm,
- 			(unsigned long)addr, pgcount, write ? FOLL_WRITE : 0,
- 			pg, NULL);
--#else
-+#elif (LINUX_VERSION_CODE < KERNEL_VERSION(5, 9, 0))
- 	ret = get_user_pages_remote(task, mm,
- 			(unsigned long)addr, pgcount, write ? FOLL_WRITE : 0,
- 			pg, NULL, NULL);
-+#else
-+	ret = get_user_pages_remote(mm,
-+			(unsigned long)addr, pgcount, write ? FOLL_WRITE : 0,
-+			pg, NULL, NULL);
- #endif
- #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0))
- 	up_read(&mm->mmap_sem);
